demos/glacial_isostatic_adjustment/2d_cylindrical/2d_cylindrical.py

```diff
@@ -0,0 +1,551 @@
+# mesh_data = pv.read("mesh/mesh_0.vtu")
+# edges = mesh_data.extract_all_edges()
+# plotter = pv.Plotter(notebook=True)
+# plotter.add_mesh(edges, color="black")
+# plotter.camera_position = "xy"
+# plotter.show(jupyter_backend="static", interactive=False)
+# -
+
+# Let's start initialising some parameters. First of all let's get the (symbolic) spatial coordinates of the mesh
+
+X = SpatialCoordinate(mesh)
+
+# Now we can set up the background profiles for the material properties. In this case the density and shear modulus vary in the vertical direction. We will approximate the series of layers using a smooth tanh function with a width of 40 km.
+
+
+# +
+# layer properties from spada et al 2011
+radius_values = [6371e3, 6301e3, 5951e3, 5701e3, 3480e3]
+density_values = [3037, 3438, 3871, 4978]
+shear_modulus_values = [0.50605e11, 0.70363e11, 1.05490e11, 2.28340e11]
+viscosity_values = [2, -2, -2, -1.698970004]  # viscosity = 1e23 * 10**viscosity_values
+# N.b. that we have modified the viscosity of the Lithosphere viscosity from
+# Spada et al 2011 because we are using coarse grid resolution
+
+
+def initialise_background_field(field, background_values, vertical_tanh_width=40e3):
+    profile = background_values[0]
+    sharpness = 1 / vertical_tanh_width
+    depth = sqrt(X[0]**2 + X[1]**2)-radius_values[0]
+    for i in range(1, len(background_values)):
+        centre = radius_values[i] - radius_values[0]
+        mag = background_values[i] - background_values[i-1]
+        profile += step_func(depth, centre, mag, increasing=False, sharpness=sharpness)
+
+    field.interpolate(profile)
+
+
+density = Function(W, name="density")
+initialise_background_field(density, density_values)
+
+shear_modulus = Function(W, name="shear modulus")
+initialise_background_field(shear_modulus, shear_modulus_values)
+
+
+# -
+
+# Let's have a quick look at the density field using pyvista. We can see that the mesh is still quite coarse, but it is able to capture the layered structure.
+
+# + tags=["active-ipynb"]
+# # Read the PVD file
+# density_file = VTKFile('density.pvd').write(density)
+# reader = pv.get_reader("density.pvd")
+# data = reader.read()[0]  # MultiBlock mesh with only 1 block
+#
+# # Create a plotter object
+# plotter = pv.Plotter(shape=(1, 1), border=False, notebook=True, off_screen=False)
+#
+# # Make a colour map
+# boring_cmap = plt.get_cmap("viridis", 25)
+# # Add the warped displacement field to the frame
+# plotter.add_mesh(
+#     data,
+#     component=None,
+#     lighting=False,
+#     show_edges=True,
+#     edge_color='grey',
+#     cmap=boring_cmap,
+#     scalar_bar_args={
+#         "title": 'Density (kg / m^3)',
+#         "position_x": 0.8,
+#         "position_y": 0.2,
+#         "vertical": True,
+#         "title_font_size": 20,
+#         "label_font_size": 16,
+#         "fmt": "%.0f",
+#         "font_family": "arial",
+#     }
+# )
+# plotter.camera_position = 'xy'
+# plotter.show()
```

Comment:
Do you use any of the interactivity in these plots? I discovered a way to ignore the big scary trame warning (screenshot below) when showing the plotter without `# plotter.show(jupyter_backend="static", interactive=False)` (as above in this file). But if there's no benefit to the interactivity, maybe we should just force it to be static?

![image](https://github.com/user-attachments/assets/c66aa4e1-850e-4dd8-8775-31abe0bdf154)

Reply:
At the moment I'm not using the interactiviity (maybe it would be nice for showing timestepping?) so happy to force it to b static